### PR TITLE
Warning when bad library used

### PR DESCRIPTION
**Pull request by remicollet**
Open for discussion.

We can also imaging to not INIT the provided classes when sapi is not CLI.
Only a initial and simple proposal.

### CODE REVIEW

**Comment by krakjoe** [comment]
I'm not sure of the effects of using a the raster render mode, it may be desirable to retain the ability to run using graphics in some cases, although it shouldn't be (and isn't currently) the default.

I'll do some digging ...

**Comment by remicollet** [comment]
> I'm not sure of the effects of using a the raster render mode, it may be desirable to retain the ability to run using graphics in some cases

OK, I have removed the 3rd commit, and only kept the stuff about checking extended version

File: wkhtmltox.c [hunk 1]
```diff
@@ -24,12 +24,15 @@
...
 #include "php.h"
 #include "php_ini.h"
+#include "SAPI.h"
 #include "ext/standard/info.h"
 #include "php_wkhtmltox.h"
 
 #include "src/pdf.h"
 #include "src/image.h"
 
+#define WKHTMLTOX_USABLE (wkhtmltopdf_extended_qt() || !strcmp(sapi_module.name, "cli"))
```
Comment (krakjoe) [inline wkhtmltox.c]: I think this condition is wrong ...

If we have extended qt then any SAPI is fine.

If we don't have extended qt then probably cli is the only reasonable SAPI - but this condition would stop it working in that circumstance.

**Comment by krakjoe** [comment]
Can you update this please ?

**Comment by remicollet** [comment]
Replaced by #5

**Pull request closed without merging**